lib/phoenix/controller/pipeline.ex

```diff
@@ -23,10 +23,25 @@ defmodule Phoenix.Controller.Pipeline do
                  &(&1 |> Map.put(:phoenix_controller, __MODULE__)
                       |> Map.put(:phoenix_action, action))
 
-        Phoenix.Endpoint.instrument conn, :phoenix_controller_call,
+        start = System.monotonic_time()
+        :telemetry.execute([:phoenix, :controller, :call, :start], %{}, %{
+          conn: conn,
+          controller: __MODULE__,
+          action: action
+        })
+        conn = Phoenix.Endpoint.instrument conn, :phoenix_controller_call,
           %{conn: conn, log_level: @phoenix_log_level}, fn ->
           phoenix_controller_pipeline(conn, action)
         end
+        stop = System.monotonic_time()
```

Comment:
Maybe these should specify the unit so that we can more clearly enforce the unit that the `duration` will have.

```suggestion
        stop = System.monotonic_time(:microsecond)
```

Reply:
Maybe the unit should be configurable? I thought that native unit makes the most sense, since anyone can easily convert it using the `:erlang.convert_time_unit`, but it is not that good from the perspective of docs, examples etc.

Reply:
+ telling the users to *always* convert the time to a desired unit is not that great from usability perspective :D

Reply:
But will Phoenix end users deal with this conversion or the telemetry attachments that they include for like publishing to prometheus?

I would think that it would mostly be the authors of latter and definitely go with `native`. If that isn't expected then yea, native may be annoying/confusing to a user of Phoenix.

Reply:
I'd say that it's the users of Phoenix. In case of Prometheus it's quite simple, because it supports only one time unit, from what I remember. But if you're sending e.g. to StatsD, you may choose whatever unit you want.

Reply:
Let's go with native for now, and we can reiterate later.

Reply:
Agreed on native.